### ISSUE TITLE
Memorize which File contains which Pruefi Tables

### ISSUE THREAD

**Issue by hf-kklein** [Hochfrequenz/kohlrahbi#148]
As of today we're re-reading the same `docx.Document`s over and over again to find out if they contain a specific pruefi. Ideally we'd reduce the re-reading and memorize which pruefi is (not) contained in which file. Even the information that a file does _not_ contain a pruefi could significantly speed up the overall scraping and I think is easier to integrate in the existing code base.

**Comment by hf-krechan** [comment]
did you not fix this issue with the PR #146 ?

**Comment by hf-kklein** [comment]
> did you not fix this issue with the PR #146 

nope. iirc the thing is, that there's still a lot of iteration over files going on when we try to do stuff like "extract all prüfis".
I cannot recall exactly right now but I think it was a 
```python
for pruefi in pruefis:
    for file in files:
        scan_file_for_pruefi(pruefi, file) # <-- takes time
```
kind of thing. we could probably speed up by alot, if instead we used something like:
```python
pruefi_file_mapping: dict[str,Path] = dict()
for file in files:
    for pruefi in get_pruefis_in_file(file):
         try:
            pruefi_file_mapping[pruefi].append(file)
         except KeyError:
            pruefi_file_mapping[pruefi] = [file]
```
you get what I mean? It should prevent re-searching the same files over and over again although we already know whether or not they contain specific pruefis.

**Comment by hf-krechan** [comment]
This issue got fixed with the PR #209
